File: AspNetMvcUnleashed/0672329980_source/Chapter14/CS/MvcApplication1/MvcApplication1/Controllers/DeleteController.cs
using System.Linq;
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class DeleteController : Controller
    {
        private MoviesDBEntities _entities = new MoviesDBEntities(); 

        public ActionResult Index()
        {
            return View(_entities.MovieSet.ToList());
        }

        [AcceptVerbs(HttpVerbs.Delete)]
        public ActionResult Delete(int id)
        {
            var movieToDelete = (from m in _entities.MovieSet
                                 where m.Id == id
                                 select m).FirstOrDefault();
            _entities.DeleteObject(movieToDelete);
            _entities.SaveChanges();
            return PartialView("Movies", _entities.MovieSet.ToList());
        }

        [ActionName("Delete")]
        public ActionResult Delete_GET(int id)
        {
            var movieToDelete = (from m in _entities.MovieSet
                                 where m.Id == id
                                 select m).FirstOrDefault();
            return View(movieToDelete);
        }

        [AcceptVerbs(HttpVerbs.Post)]
        [ActionName("Delete")]
        public ActionResult Delete_POST(int id)
        {
            var movieToDelete = (from m in _entities.MovieSet
                                 where m.Id == id
                                 select m).FirstOrDefault();
            _entities.DeleteObject(movieToDelete);
            _entities.SaveChanges();
            return RedirectToAction("Index");
        }

    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter14/VB/MvcApplication1/MvcApplication1/Controllers/DeleteController.vb
Public Class DeleteController
    Inherits Controller

    Private _entities As New MoviesDBEntities()

    Public Function Index() As ActionResult
        Return View(_entities.MovieSet.ToList())
    End Function

    <AcceptVerbs(HttpVerbs.Delete)> _
    Public Function Delete(ByVal id As Integer) As ActionResult
        Dim movieToDelete = (From m In _entities.MovieSet _
                             Where m.Id = id _
                             Select m).FirstOrDefault()
        _entities.DeleteObject(movieToDelete)
        _entities.SaveChanges()
        Return PartialView("Movies", _entities.MovieSet.ToList())
    End Function

    <ActionName("Delete")> _
    Public Function Delete_GET(ByVal id As Integer) As ActionResult
        Dim movieToDelete = (From m In _entities.MovieSet _
                             Where m.Id = id _
                             Select m).FirstOrDefault()
        Return View(movieToDelete)
    End Function

    <AcceptVerbs(HttpVerbs.Post), ActionName("Delete")> _
    Public Function Delete_POST(ByVal id As Integer) As ActionResult
        Dim movieToDelete = (From m In _entities.MovieSet _
                             Where m.Id = id _
                             Select m).FirstOrDefault()
        _entities.DeleteObject(movieToDelete)
        _entities.SaveChanges()
        Return RedirectToAction("Index")
    End Function

End Class
